app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddCyclewayPartSurface.kt

```diff
@@ -29,7 +37,15 @@ class AddCyclewayPartSurface(o: OverpassMapDataAndGeometryApi, r: ResurveyInterv
 
     override fun createForm() = AddPathSurfaceForm()
 
-    override fun applyAnswerTo(answer: String, changes: StringMapChangesBuilder) {
-        changes.updateWithCheckDate("cycleway:surface", answer)
-    }
+    override fun applyAnswerTo(answer: DetailSurfaceAnswer, changes: StringMapChangesBuilder) {
+        when(answer) {
+            is SurfaceAnswer -> {
+                changes.updateWithCheckDate("cycleway:surface", answer.value)
+                changes.deleteIfExists("source:cycleway:surface")
```

Comment:
Why this line? Should it maybe be "cycleway:surface:note"?

Reply:
For now it is for consistency with deleting `source:surface` tag in basic quest. I am not entirely sure about it.

And yes, deleting `cycleway:surface:note` may be a good idea but I am also not 100% sure about it. Either way I will leave it for later.

Reply:
Ah, understood. It's the "Bing removal". Nevermind then, it's all the same to me.

Since this is a resurvey quest, I would say it is quite necessary to delete the `cycleway:surface:note` if the selection is not generic, otherwise data garbage is left behind.

Reply:
Added removal of `*surface:note` tags.